### ISSUE TITLE
stop / frequency_penalty / presence_penalty silently ignored with no client signal

### ISSUE THREAD

**Issue by dpalmqvist** [motsognirr/olmlx#341]
## Summary

`stop` sequences and the `frequency_penalty` / `presence_penalty` knobs are dropped before generation. The server logs a `WARNING` but the response still claims `finish_reason: "stop"` and looks normal, so clients can't tell the parameter was a no-op.

## Repro

```bash
curl -s http://localhost:11434/v1/chat/completions -d '{
  "model": "mlx-community/Qwen3-4B-4bit",
  "messages": [{"role": "user", "content": "Just literally repeat: A B C D E F G"}],
  "chat_template_kwargs": {"enable_thinking": false},
  "stop": ["D"],
  "max_tokens": 50
}'
```

Returns `"A B C D E F G"` with `finish_reason: "stop"` (i.e. the stop sequence "D" was ignored — generation continued past it).

Server log:
```
WARNING olmlx.engine.inference: [a76a75f6] stop sequences not supported by mlx-lm >= 0.30.7; ignored
```

## Source

`olmlx/engine/inference.py:1200-1207`:

```python
if "stop" in options:
    logger.warning("stop sequences not supported by mlx-lm >= 0.30.7; ignored")

for penalty_key in ("frequency_penalty", "presence_penalty"):
    if penalty_key in options:
        logger.warning(
            "%s not supported by mlx-lm >= 0.30.7; ignored", penalty_key
        )
```

The same applies to `/api/chat`, `/api/generate`, and `/v1/messages` (Anthropic passes `stop_sequences` through `build_inference_options`).

## Why this matters

Clients ask for `stop` to guarantee they get a bounded block — e.g. "stop at `\n\n`" when extracting a single paragraph, or "stop at `</answer>`" for structured prompts. A silent no-op breaks those callers in ways that are hard to attribute to the server.

## Options

1. **Reimplement on top of `stream_generate`.** Both `stop` and the repetition penalties can be enforced in a small wrapper around mlx-lm's token-by-token loop (check the rolling decoded suffix against each `stop`; apply the penalty logits processor before sampling). The penalties already exist as `mlx-lm` sampler utilities. This is the right long-term fix.
2. **Fail loudly** if the operator opts in: a config flag (`OLMLX_STRICT_OPTIONS=true`) that turns the warnings into 400s. Cheap, surfaces the gap to clients.
3. **At minimum**, advertise the gap: include a `warnings` array in the response (`{"warnings": ["stop sequences ignored: not supported by mlx-lm >= 0.30.7"]}`) so SDK clients can log it. OpenAI itself doesn't have this field, but it's harmless to add for unrecognized clients and doesn't break the schema.

## Impact

Spec-conforming OpenAI / Ollama / Anthropic clients that depend on `stop` get longer output than expected with no error. Same for callers using `frequency_penalty` / `presence_penalty` to control repetition.
